test/integration/puppetlabs/services/jruby/class_info_test.clj

```diff
@@ -221,7 +224,105 @@
 
           (testing "non-existent environment"
             (is (nil? (get-class-info-for-env "bogus-env"))))
-
-        (finally
-          (.terminate jruby-puppet)
-          (.terminate container))))))
+          (testing "(PUP-5744) non-JSON safe default_literals omitted"
+            (let [env-5-dir (env-dir "env5")
+                  _ (create-env-conf env-5-dir "modulepath=\n")
+                  foo-manifest (.getAbsolutePath (fs/file env-5-dir
+                                                          "manifests"
+                                                          "foo.pp"))]
+              (create-file foo-manifest
+                           (str
+                            "class foo (\n"
+                            "  Regexp $some_regex = /^.*/,\n"
+                            "  Default $some_default = default,\n"
+                            "  Hash $some_hash = { 1 => 2, "
+                            "\"two\" => 3},\n"
+                            "  Hash $some_nested_hash = { \"one\" => 2, "
+                            "\"two\" => { 3 => 4 }},\n"
+                            "  Hash $another_nested_hash = { \"one\" => 2, "
+                            "\"two\" => { \"three\" => 4 }},\n"
+                            "  Array $some_array = [ 1, /^*$/ ],\n"
+                            "  Array $some_nested_array = [ 1, [ 2, "
+                            "default ] ],\n"
+                            "  Array $another_nested_array = [ 1, [ 2, 3 ] ]\n"
+                            "){}"))
```

Comment:
Yeah, good idea.  Some of the "\n"s that I had here were just cosmetic formatting in case you wanted to look at the file on disk for troubleshooting.  Puppet would probably parse these files okay without them.

I agree with the idea about the ks function, though.  Might do that the next time I have a need to add on stuff like this if someone else doesn't get to it first.